[TensorExpr] Fix 'isIntegralType is deprecated' warning.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47403 [TensorExpr] Fix 'isIntegralType is deprecated' warning.**
* #47402 [TensorExpr] Run constant pooling in fusion groups to dedupe constants.

Differential Revision: [D24740958](https://our.internmc.facebook.com/intern/diff/D24740958)

Commit:
[TensorExpr] Fix 'isIntegralType is deprecated' warning.

[ghstack-poisoned]

eellison commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -413,7 +413,7 @@ ExprHandle TensorExprKernel::constant(const torch::jit::Value* v) {
 ExprHandle promoteIntegerToFloat(const ExprHandle& e) {
   auto scalarType = static_cast<c10::ScalarType>(e.dtype().scalar_type());
-  if (!c10::isIntegralType(scalarType)) {
+  if (!c10::isIntegralType(scalarType, /* includeBool = */ false)) {
```
This is going to conflict with : https://github.com/pytorch/pytorch/pull/47403/ where i set it to `includeBool` to true (bc in eager these ops also promote bools)

Commit:
Update on "[TensorExpr] Fix 'isIntegralType is deprecated' warning."


Differential Revision: [D24740958](https://our.internmc.facebook.com/intern/diff/D24740958)

[ghstack-poisoned]

ZolotukhinM commented:
Abandoning in favor of #47374.
